src/cbmc/cbmc_parse_options.cpp

```diff
@@ -110,6 +108,16 @@ void cbmc_parse_optionst::get_command_line_options(optionst &options)
     exit(CPROVER_EXIT_USAGE_ERROR);
   }
 
+  if(cmdline.isset("paths"))
+    options.set_option("paths", true);
+
+  if(cmdline.isset("goto-priority"))
+    options.set_option("goto-priority", cmdline.get_value("goto-priority"));
+  else if(options.get_bool_option("paths"))
+    options.set_option("goto-priority", "jump");
```

Comment:
Maybe use an enum for this in `optionst`?

Reply:
@smowton I don't understand. How would I be able to call `optionst::set_option()` with an enum as an argument? The only way I can think of would be to have a dedicated `optionst::set_goto_priority()` function and have the goto_priority be a field inside `optionst`, is this what you meant? Or am I misunderstanding?

Reply:
No, I meant just using `optionst`'s `set_option(unsigned int)` overload and corresponding `get_unsigned_int_option`. It's not a lot better since `optiont` will stringify and then `atoi` it internally and doesn't enforce type, but at least a `switch` on some `enum goto_priorityt` will yield incomplete switch-case warnings from your compiler if and when it grows a third option.